src/urllib3/util/url.py

```diff
@@ -128,9 +128,27 @@ def request_uri(self) -> str:
 
         return uri
 
+    @property
+    def authority(self) -> Optional[str]:
+        """
+        Network location including userinfo, host and port.
```

Comment:
Not sure if we should describe it as the "network location"? Honestly it's still strange to me that netloc included authentication.

Let's also leave a reference to where this component is defined (ie RFC 3986 3.2)